File: native/Marketplace/app/src/main/java/com/example/marketplace/models/Listing.kt
package com.example.marketplace.models

import android.net.Uri
import com.example.marketplace.enums.BodyStyle
import com.example.marketplace.enums.Condition
import com.example.marketplace.enums.EmissionStandard
import com.example.marketplace.enums.FuelType
import java.util.Date

data class Listing (
    var listingId: Int = 0,
    var title: String = "",
    var description: String = "",
    var price: Int = 0,
    var condition: Condition = Condition.New,
    var brand: String = "",
    var model: String = "",
    var fuelType: FuelType = FuelType.Diesel,
    var bodyStyle: BodyStyle = BodyStyle.Sedan,
    var colour: String = "",
    var manufactureYear: Int = 2000,
    var mileage: Int = 0,
    var emissionStandard: EmissionStandard = EmissionStandard.Non_euro,
    var sellerId: Int = 0,
    var sellerName: String = "",
    var country: String = "",
    var city: String = "",
    var createdAt: Date = Date(),
    var imageUri: Uri? = null
) {
    fun replaceListingData(listing: Listing) {
        this.title = listing.title
        this.description = listing.description
        this.price = listing.price
        this.condition = listing.condition
        this.brand = listing.brand
        this.model = listing.model
        this.fuelType = listing.fuelType
        this.bodyStyle = listing.bodyStyle
        this.colour = listing.colour
        this.manufactureYear = listing.manufactureYear
        this.mileage = listing.mileage
        this.emissionStandard = listing.emissionStandard
        this.imageUri = listing.imageUri
    }

    fun isValidListing(): Boolean {
        if (title.isEmpty()) return false
        if (price == 0) return false
        if (brand.isEmpty()) return false
        if (model.isEmpty()) return false
        if (mileage == 0) return false

        return true
    }
}

File: server/CarMarketplace/CarMarketplace/Models/Listing.cs
namespace CarMarketplace.Models
{
    public class Listing
    {
        public int ListingId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public int Price { get; set; } = 1;
        public int Condition { get; set; } = 0;
        public string Brand { get; set; } = string.Empty;
        public string Model { get; set; } = string.Empty;
        public int FuelType { get; set; } = 0;
        public int BodyStyle { get; set; } = 0;
        public string Colour { get; set; } = string.Empty;
        public int ManufactureYear { get; set; } = 2000;
        public int Mileage { get; set;} = 0;
        public int EmissionStandard { get; set; } = 0;
        public int SellerId { get; set; } = 0;
        public string SellerName { get; set; } = string.Empty;
        public string Country { get; set; } = string.Empty;
        public string City { get; set; } = string.Empty;
        public DateOnly CreatedAt { get; set; }
        public string? ImagePath { get; set; }
    }
}
